stdlib/set.ml

```diff
@@ -609,15 +625,169 @@ module Make(Ord: OrderedType) =
 
     let to_rev_seq c = rev_seq_of_enum_ (snoc_enum c End)
 
+    (* [enum_from_aux low s c] constructs an enumeration whose elements are:
+       1- the elements [x] of the tree [s] such that [low <= x] holds,
+       followed with 2- the elements of the enumeration [c]. *)
+    let rec enum_from_aux low s c = match s with
+      | Empty -> c
+      | Node {l; r; v; _} ->
+          begin match Ord.compare v low with
+            | 0 -> More (v, r, c)
+            | n when n<0 -> enum_from_aux low r c
+            | _ -> enum_from_aux low l (More (v, r, c))
+          end
+
+    let[@inline] enum_from low s =
+      enum_from_aux low s End
+
     let to_seq_from low s =
-      let rec aux low s c = match s with
-        | Empty -> c
-        | Node {l; r; v; _} ->
-            begin match Ord.compare v low with
-              | 0 -> More (v, r, c)
-              | n when n<0 -> aux low r c
-              | _ -> aux low l (More (v, r, c))
-            end
-      in
-      seq_of_enum_ (aux low s End)
+      seq_of_enum_ (enum_from low s)
+
+    module Enum = struct
+
+      (* The type [enumeration] is the same as above: *)
+
+      (* type enumeration = End | More of elt * t * enumeration *)
```

Comment:
It looks like you are trying to modify the existing code in the least invasive way possible, but the result is a bit confusing, with a submodule for enumerations whose main definition is sitting outside, somewhere else in the code. If it is easy to do so, I would support moving the type definition inside the `Enum` module, and then re-exporting the outside-the-module functions after its definition (instead of keeping them above in their previous place, as you are doing here).